src/asmcnc/skavaUI/screen_go.py

```diff
@@ -597,49 +597,38 @@ def _start_running_job(self):
+                modified_job_gcode.append("M56")  # append cleaned up gcode to object
+                self.jd.job_recovery_offset += 1 # number of lines added to start of job
 
-        # If job recovery is being performed, use job recovery gcode instead
-        if job_recovery_active:
             # Turn vac on if spindle gets turned on during job
-            if ((str(self.jd.job_recovery_gcode).count("M3") > str(self.jd.job_recovery_gcode).count("M30")) or (
-                    str(self.jd.job_recovery_gcode).count("M03") > 0)) and self.m.stylus_router_choice != 'stylus':
+            if ((str(job_to_modify).count("M3") > str(job_to_modify).count("M30")) or (
+                    str(job_to_modify).count("M03") > 0)) and self.m.stylus_router_choice != 'stylus':
                 modified_job_gcode.append("AE")  # turns vacuum on
                 modified_job_gcode.append("G4 P2")  # sends pause command
-                modified_job_gcode.extend(self.jd.job_recovery_gcode)
+                modified_job_gcode.extend(job_to_modify)
                 modified_job_gcode.append("G4 P2")  # sends pause command, 2 seconds
                 modified_job_gcode.append("AF")  # turns vac off
                 self.jd.job_recovery_offset += 2 # number of lines added to start of job
             else:
-                modified_job_gcode.extend(self.jd.job_recovery_gcode)
-        else:
-            # Turn vac on if spindle gets turned on during job
-            if ((str(self.jd.job_gcode).count("M3") > str(self.jd.job_gcode).count("M30")) or (
-                    str(self.jd.job_gcode).count("M03") > 0)) and self.m.stylus_router_choice != 'stylus':
-                modified_job_gcode.append("AE")  # turns vacuum on
-                modified_job_gcode.append("G4 P2")  # sends pause command
-                modified_job_gcode.extend(self.jd.job_gcode)
-                modified_job_gcode.append("G4 P2")  # sends pause command, 2 seconds
-                modified_job_gcode.append("AF")  # turns vac off
-                self.jd.job_recovery_offset += 2 # number of lines added to start of job
-            else:
-                modified_job_gcode.extend(self.jd.job_gcode)
+                modified_job_gcode.extend(job_to_modify)
+
+            # Spindle command??
+            if self.lift_z_on_job_pause and self.m.fw_can_operate_zUp_on_pause():  # extra 'and' as precaution
+                modified_job_gcode.append("M56 P0")  # append cleaned up gcode to object
+
+            return modified_job_gcode
 
-        # Spindle command??
-        if self.lift_z_on_job_pause and self.m.fw_can_operate_zUp_on_pause():  # extra 'and' as precaution
-            modified_job_gcode.append("M56 P0")  # append cleaned up gcode to object
+        # Check if job recovery is being performed
+        if self.jd.job_recovery_filepath == self.jd.filename and self.jd.job_recovery_selected_line != -1:
+            modified_job_gcode = prep_gcode(self.jd.job_recovery_gcode)
```

Comment:
I would have modified_job_gcode in the inner prep_gcode function renamed, as I think both versions of the variable might exist within the same scope as prep_gcode is a nested function. I think it would be ok (as it ultimately ends up being the same list anyway), but might still be a good idea to rename :).